history/taskana-simplehistory-provider/src/test/java/acceptance/events/task/CreateHistoryEventOnTaskDeletionAccTest.java

```diff
@@ -0,0 +1,157 @@
+        DefaultTestEntities.defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    task1 = createTask().buildAndStore(taskService);
+    task2 = createTask().state(TaskState.COMPLETED).buildAndStore(taskService);
+    task3 = createTask().state(TaskState.COMPLETED).buildAndStore(taskService);
+  }
+
+  @WithAccessId(user = "admin")
+  @Test
+  void should_CreateDeleteHistoryEvent_When_TaskIsDeleted() throws Exception {
+    historyService.deleteHistoryEventsByTaskIds(List.of(task1.getId()));
+    TaskHistoryQueryMapper taskHistoryQueryMapper = getHistoryQueryMapper();
+    List<TaskHistoryEvent> events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(task1.getId()));
+    historyService.createTaskHistoryQuery().taskIdIn(task1.getId());
+    assertThat(events).isEmpty();
+
+    taskService.forceDeleteTask(task1.getId());
+
+    events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(task1.getId()));
+    assertThat(events).hasSize(1);
+    assertDeleteHistoryEvent(events.get(0).getId(), TaskState.READY.toString(), "DELETED", "admin");
+  }
+
+  @WithAccessId(user = "admin")
+  @Test
+  void should_CreateDeleteHistoryEvents_When_MultipleTasksAreDeleted() throws Exception {
+    List<String> taskIds = List.of(task2.getId(), task3.getId());
+    historyService.deleteHistoryEventsByTaskIds(taskIds);
+    TaskHistoryQueryMapper taskHistoryQueryMapper = getHistoryQueryMapper();
+    List<TaskHistoryEvent> events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl)
+                historyService.createTaskHistoryQuery().taskIdIn(taskIds.toArray(new String[0])));
+    assertThat(events).isEmpty();
+
+    taskService.deleteTasks(taskIds);
+
+    events =
+        taskHistoryQueryMapper.queryHistoryEvents(
+            (TaskHistoryQueryImpl)
+                historyService.createTaskHistoryQuery().taskIdIn(taskIds.toArray(new String[0])));
+    assertThat(events)
+        .extracting(TaskHistoryEvent::getTaskId)
+        .containsExactlyInAnyOrderElementsOf(taskIds);
+    for (TaskHistoryEvent event : events) {
+      assertDeleteHistoryEvent(event.getId(), TaskState.COMPLETED.toString(), "DELETED", "admin");
+    }
+  }
+
+  TaskHistoryQueryMapper getHistoryQueryMapper()
+      throws NoSuchFieldException, IllegalAccessException {
+    Field sessionManagerField = TaskanaHistoryEngineImpl.class.getDeclaredField("sessionManager");
+    sessionManagerField.setAccessible(true);
+    SqlSessionManager sqlSessionManager =
+        (SqlSessionManager) sessionManagerField.get(taskanaHistoryEngine);
+
+    return sqlSessionManager.getMapper(TaskHistoryQueryMapper.class);
+  }
+
+  private void assertDeleteHistoryEvent(
+      String eventId, String expectedOldValue, String expectedNewValue, String expectedUser)
```

Comment:
Could you please rename the parameters to be more readable? For example "expectedNewTaskState" instead of "expectedNewValue"?